core/contracts/Store.sol

```diff
@@ -0,0 +1,112 @@
+/*
+  Store implementation.
+ 
+  An implementation of StoreInterface with a fee per second and withdraw functions for the owner.
+*/
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "./StoreInterface.sol";
+import "./Withdrawable.sol";
+import "./FixedPoint.sol";
+import "./MultiRole.sol";
+
+
+// An implementation of StoreInterface that can accept Oracle fees in ETH or any arbitrary ERC20 token.
+contract Store is StoreInterface, MultiRole {
+
+    using SafeMath for uint;
+    using FixedPoint for FixedPoint.Unsigned;
+    using FixedPoint for uint;
+
+    enum Roles {
+        Governance
+    }
+
+    uint private fixedOracleFeePerSecond; // Percentage of 10^18. E.g., 1e18 is 100% Oracle fee.
+    uint private constant FP_SCALING_FACTOR = 10**18;
+
+    uint private weeklyDelayFee;
+    mapping(address => FixedPoint.Unsigned) private finalFees;
+    uint private constant SECONDS_PER_WEEK = 604800;
+
+    // TODO(roz): Used to make doubly sure that roles are initialized only once. 
+    // Figure out what's going wrong with coverage to necessitate this hack.
+    bool private rolesInitialized;
+
+    constructor() public {
+        initializeRolesOnce();
+    }
+
+    function payOracleFees() external payable {
+        require(msg.value > 0);
+    }
+
+    function payOracleFeesErc20(address erc20Address) external {
+        IERC20 erc20 = IERC20(erc20Address);
+        uint authorizedAmount = erc20.allowance(msg.sender, address(this));
+        require(authorizedAmount > 0);
+        require(erc20.transferFrom(msg.sender, address(this), authorizedAmount));
+    }
+
+    // Sets a new Oracle fee per second
+    function setFixedOracleFeePerSecond(FixedPoint.Unsigned calldata newOracleFee) 
+        external 
+        onlyRoleHolder(uint(Roles.Governance)) 
+    {
+        // Oracle fees at or over 100% don't make sense.
+        require(newOracleFee.isLessThan(1));
+        fixedOracleFeePerSecond = newOracleFee.value;
+        emit SetFixedOracleFeePerSecond(newOracleFee);
+    }
+
+    function setFinalFee(address currency, uint finalFee) 
+        external 
+        onlyRoleHolder(uint(Roles.Governance))
```

Comment:
drive-by:
I think there's some chance that this method may be called much more frequently than `setFixedOracleFeePerSecond`, because IIRC, the final fee adjusts for the relative prices of margin currencies, which could be very volatile.

Thoughts on allowing a role separate from the governance role to set fixed fees?

Reply:
You're probably right about this being accessed more. Wouldn't it still be a governance vote though?

Reply:
Good idea!

Reply:
It might be by governance vote at some point in the future. But in the meantime, we would be changing this with a private key. We would probably want a way to delegate this to a "hot" key. I think this same setup might make sense for the setter of the WeeklyDelayFee and the FixedOracleFee as well. We want our "super root" key to be used as little as possible.